### DISCUSSION TITLE
[vi-mode] how to start in normal mode?

### DISCUSSION THREAD

**Discussion by pogopaule** [ohmyzsh/ohmyzsh#10674] [Help] [answered]
How can I configure the vi-mode plugin to start the prompt in normal mode. Now it starts every new prompt in insert mode. But like in vim, it makes more sense to me to start in normal mode.

**Comment by carlosala** [comment]
There's no config to do that.
Maybe you can try to emulate an `ESC` click at the bottom of your `zshrc` file. You can also do a pull request submitting that change, we'll be happy to review it and merge it!

**Comment by mcornella** [comment, marked as answer]
You could leverage `zle-line-init` to set the keymap to `vicmd`. This would work without overriding previous zle-line-init hooks:
```zsh
autoload -Uz add-zle-hook-widget
add-zle-hook-widget line-init vi-cmd-mode
```

EDIT: [source](https://zsh.sourceforge.io/Doc/Release/Zsh-Line-Editor.html#index-zle_002dline_002dinit).

**Reply by pogopaule** [reply]
I put the two lines at the end of my .zshrc, and restarted my terminal but the line still starts in insert mode.

**Reply by mcornella** [reply]
What zsh version are you using? (run `echo $ZSH_VERSION`). `add-zle-hook-widget` was added in 5.3, so you might need a different solution.

**Reply by mcornella** [reply]
This is a quick hack that might work in this case
```zsh
functions[zle-line-init]="zle vi-cmd-mode; ${functions[zle-line-init]}"
```

**Reply by mcornella** [reply]
Apologies, the original code was broken, it should be `line-init` passed to `add-zle-hook-widget`, not `zle-line-init`. Please try it again https://github.com/ohmyzsh/ohmyzsh/discussions/10674#discussioncomment-2148831

**Reply by rwmitchell** [reply]
you must have changed it while i was trying it...wondered why i typed it wrong.

still didn't work for me, but was trying it in a current shell.

it does work if drop back to a basic zsh setup, but not with my full omz+p10k config

**Reply by mcornella** [reply]
> Apologies, the original code was broken, it should be `line-init` passed to `add-zle-hook-widget`, not `zle-line-init`. Please try it again [#10674 (comment)](https://github.com/ohmyzsh/ohmyzsh/discussions/10674#discussioncomment-2148831)

Not true, it works both ways. So if it doesn't work for you try https://github.com/ohmyzsh/ohmyzsh/discussions/10674#discussioncomment-2159771.

**Reply by rwmitchell** [reply]
I did say it didn't work me, I'm assuming something I've customized is breaking it.  The 'quick' hack does indeed work, however.
I'm running: 5.8.0.2-dev which is well beyond 5.3.

**Reply by mcornella** [reply]
I was specifically correcting my previous message, not yours. I have no idea why `add-zle-hook-widget` does not work in your case, you could try to add a test function with `zle -M testfunc` in its body (as an example, `zle -M` in this case is used in place of `echo` so it doesn't mess up the prompt).

**Reply by rwmitchell** [reply]
Thanks, but my cares are over at this point.  I was amused by the concept, enough to want to try it and now have a 'vicmd' function to flip it on, but don't think I'll be using it very often.

Still, I'm impressed it can be done and you figured it out.

**Reply by pogopaule** [reply]
> This is a quick hack that might work in this case
> 
> ```shell
> functions[zle-line-init]="zle vi-cmd-mode; ${functions[zle-line-init]}"
> ```

This works! Thank you!
P.S.: The other (corrected) solution does not work for me either. I use version 5.8.

**Reply by pogopaule** [reply]
Damn, just saw that there is already an answer...didn't use my google-fu correctly :disappointed: 
https://unix.stackexchange.com/questions/438307/zsh-start-new-prompt-in-command-mode-vi-mode
